File: Jack.Business/Kit.vb
Public Class Kit
    Implements ICrud(Of Model.Kit, Integer)

    Public Sub New()

    End Sub

    Public Function Delete(oTipo As Model.Kit) As Boolean Implements ICrud(Of Model.Kit, Integer).Delete
        Dim oDados As Data.Kit = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.Kit
            blnRetorno = oDados.Delete(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno

    End Function

    Public Function Find(Identifier As Integer) As Model.Kit Implements ICrud(Of Model.Kit, Integer).Find

        Dim oDados As Data.Kit = Nothing
        Dim lstRetorno As IList(Of Model.Kit) = Nothing

        Try
            oDados = New Data.Kit
            lstRetorno = oDados.LoadAll()
        Catch ex As Exception
            lstRetorno = Nothing
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return lstRetorno

    End Function

    Public Function Insert(oTipo As Model.Kit) As Boolean Implements ICrud(Of Model.Kit, Integer).Insert
        Dim oDados As Data.Kit = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.Kit
            blnRetorno = oDados.Insert(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno
    End Function

    Public Function LoadAll() As IList(Of Model.Kit) Implements ICrud(Of Model.Kit, Integer).LoadAll
        Dim oDados As Data.Kit = Nothing
        Dim lstRetorno As IList(Of Model.Kit) = Nothing

        Try
            oDados = New Data.Kit
            lstRetorno = oDados.LoadAll()
        Catch ex As Exception
            lstRetorno = Nothing
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return lstRetorno

    End Function

    Public Function Update(oTipo As Model.Kit) As Boolean Implements ICrud(Of Model.Kit, Integer).Update
        Dim oDados As Data.Kit = Nothing
        Dim blnRetorno As Boolean = False

        Try
            oDados = New Data.Kit
            blnRetorno = oDados.Update(oTipo)
        Catch ex As Exception
            blnRetorno = False
            Throw ex
        Finally
            oDados = Nothing
        End Try

        Return blnRetorno
    End Function
End Class

File: Jack.Business/Kit.cs
using System;
using System.Collections.Generic;

namespace Jack.Business
{
    public class Kit : ICrud<Model.Kit, int>
    {

        public Kit()
        {
        }

        public bool Delete(Model.Kit oTipo)
        {
            Data.Kit oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Data.Kit();
                blnRetorno = oDados.Delete(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;

        }

        public Model.Kit Find(int Identifier)
        {

            Data.Kit oDados = null;
            IList<Model.Kit> lstRetorno = null;

            try
            {
                oDados = new Data.Kit();
                lstRetorno = oDados.LoadAll();
            }
            catch (Exception ex)
            {
                lstRetorno = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return lstRetorno;

        }

        public bool Insert(Model.Kit oTipo)
        {
            Data.Kit oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Data.Kit();
                blnRetorno = oDados.Insert(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;
        }

        public IList<Model.Kit> LoadAll()
        {
            Data.Kit oDados = null;
            IList<Model.Kit> lstRetorno = null;

            try
            {
                oDados = new Data.Kit();
                lstRetorno = oDados.LoadAll();
            }
            catch (Exception ex)
            {
                lstRetorno = null;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return lstRetorno;

        }

        public bool Update(Model.Kit oTipo)
        {
            Data.Kit oDados = null;
            bool blnRetorno = false;

            try
            {
                oDados = new Data.Kit();
                blnRetorno = oDados.Update(oTipo);
            }
            catch (Exception ex)
            {
                blnRetorno = false;
                throw ex;
            }
            finally
            {
                oDados = null;
            }

            return blnRetorno;
        }
    }
}
